deltacat/utils/resources.py

```diff
@@ -0,0 +1,51 @@
+import ray
+from typing import Dict, Any
+from dataclasses import dataclass
+from deltacat import logs
+import logging
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+
+@dataclass
+class ClusterUtilization:
+    def __init__(
+        self, cluster_resources: Dict[str, Any], available_resources: Dict[str, Any]
+    ):
+        used_resources = {}
+
+        for key in cluster_resources:
+            if (
+                isinstance(cluster_resources[key], float)
+                or isinstance(cluster_resources[key], int)
+            ) and key in available_resources:
+                used_resources[key] = cluster_resources[key] - available_resources[key]
+
+        self.total_memory_bytes = cluster_resources.get("memory")
+        self.used_memory_bytes = used_resources.get("memory")
+        self.total_cpu = cluster_resources.get("CPU")
+        self.used_cpu = used_resources.get("CPU")
+        self.total_object_store_memory_bytes = cluster_resources.get(
+            "object_store_memory"
+        )
+        self.used_object_store_memory_bytes = used_resources.get("object_store_memory")
+        self.used_resources = used_resources
+
+    @staticmethod
+    def get_current_cluster_utilization():
+        cluster_resources = ray.cluster_resources()
+        available_resources = ray.available_resources()
+
+        return ClusterUtilization(
+            cluster_resources=cluster_resources, available_resources=available_resources
+        )
+
+
+def log_current_cluster_utlization(log_identifier: str):
+    cluster_utilization = ClusterUtilization.get_current_cluster_utilization()
```

Comment:
For sanity and insights into long-term trends, I'd also like to keep durable tabs on how much time this is taking to collect across successive versions of Ray and DeltaCAT:
```
cluster_utilization, latency = timed_invocation(ClusterUtilization.get_current_cluster_utilization)
logger.info(f"Cluster Utilization Collection Latency: {latency}s")
```

Reply:
I believe this is fetching cached data from GCS, so it should not involve lot of latency. Do you think there is any value in tracking this?

Reply:
It's better to validate than assume when doing additional operations like this to collect telemetry that isn't necessary for E2E functionality in production environments. Ideally, these types of telemetry operations should be disabled automatically in production if their latency exceeds a given threshold, but this is a basic starting point to ensure that your assumption is correct today, and remains correct going forward.

Reply:
Okay, that makes sense. One scenario I can think of is, when cluster is overloaded and GCS isn't easily accessible. I will log the latency and later (next PR) capture overall telemetry latency in the audit. FYI, this call takes `0.5857944488525391` milliseconds.